pkg/kubectl/cmd/set/set_serviceaccount.go

```diff
@@ -0,0 +1,207 @@
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	"k8s.io/kubernetes/pkg/kubectl/resource"
+	"k8s.io/kubernetes/pkg/util/i18n"
+)
+
+var (
+	serviceaccountResources = `
+	replicationcontroller (rc), deployment (deploy), daemonset (ds), job, replicaset (rs), statefulset`
+
+	serviceaccountLong = templates.LongDesc(`
+	Update ServiceAccount of pod template resources.
+
+	Possible resources (case insensitive) can be: 
+	` + serviceaccountResources)
+
+	serviceaccountExample = templates.Examples(`
+	# Set Deployment nginx-deployment's ServiceAccount to serviceaccount1
+	$kubectl set service-account deployment nginx-deployment serviceaccount1
+
+	# Short form of the above command
+	$kubectl set sa deploy nginx-deployment serviceaccount1
+
+	# Print result in yaml format of updated nginx deployment from local file, without hitting apiserver
+	$kubectl set sa -f nginx-deployment.yaml serviceaccount1 --local --dry-run -o yaml
+	`)
+)
+
+// serviceAccountConfig encapsulates the data required to perform the operation.
+type serviceAccountConfig struct {
+	fileNameOptions resource.FilenameOptions
+
+	mapper                 meta.RESTMapper
+	encoder                runtime.Encoder
+	decoder                runtime.Decoder
+	out                    io.Writer
+	err                    io.Writer
+	dryRun                 bool
+	shortOutput            bool
+	all                    bool
+	record                 bool
+	output                 string
+	changeCause            string
+	local                  bool
+	categoryExpander       resource.CategoryExpander
+	clientMapper           resource.ClientMapper
+	typer                  runtime.ObjectTyper
+	namespace              string
+	enforceNamespace       bool
+	args                   []string
+	_print                 func(obj runtime.Object) error
+	updatePodSpecForObject func(runtime.Object, func(*api.PodSpec) error) (bool, error)
+}
+
+// NewCmdServiceAccount returns the "set service-account" command.
+func NewCmdServiceAccount(f cmdutil.Factory, out, err io.Writer) *cobra.Command {
+	saConfig := &serviceAccountConfig{
+		out: out,
+		err: err,
+	}
+
+	cmd := &cobra.Command{
+		Use:     "service-account (-f FILENAME | TYPE NAME) SERVICE_ACCOUNT",
+		Aliases: []string{"sa"},
+		Short:   i18n.T("Update ServiceAccount of a resource"),
+		Long:    serviceaccountLong,
+		Example: serviceaccountExample,
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(saConfig.Complete(f, cmd, args))
+			cmdutil.CheckErr(saConfig.Validate())
+			cmdutil.CheckErr(saConfig.Run())
+		},
+	}
+	cmdutil.AddPrinterFlags(cmd)
+
+	usage := "identifying the resource to get from a server."
+	cmdutil.AddFilenameOptionFlags(cmd, &saConfig.fileNameOptions, usage)
+	cmd.Flags().BoolVar(&saConfig.all, "all", false, "Select all resources in the namespace of the specified resource types")
+	cmd.Flags().BoolVar(&saConfig.local, "local", false, "If true, set image will NOT contact api-server but run locally.")
```

Comment:
And this one.  Or something.  We see it all over as the builders are getting constructed.